PISN & type Ia models - On hold

jpkneller opened the pull request:
New pisn model classes and Jupyter workbook

jpkneller committed:
Add files via upload

New pisn model classes and Jupyter workbook

jpkneller committed:
Rename doc/source/nb/Wright_2017.ipynb to doc/source/nb/pisn/Wright_2017.ipynb

jpkneller committed:
Add files via upload

pisn classes and loaders

jpkneller committed:
TypeIa SN models

jpkneller committed:
Add files via upload

jpkneller committed:
Add files via upload

jpkneller committed:
Rename doc/source/nb/TypeIa.ipynb to doc/source/nb/TypeIa/TypeIa.ipynb

JostMigenda committed:
Merge pull request #416 from SNEWS2/jpkneller-TypeIa

TypeIa SN models

JostMigenda commented on python/snewpy/models/pisn_loaders.py:
```diff
@@ -0,0 +1,124 @@
+
+        initial_spectra = {}
+        for flavor in ThreeFlavor:
+            initial_spectra[flavor] = self.interpolation[flavor]((t, E)) / (u.MeV * u.s)
```
If both `t` and `E` are multi-element arrays (e.g. if we do `model.get_flux(times, energies, d)` as demonstrated in `FlavorTransformation.ipynb`), this will fail.

The fix could be something like:

```python
        t = u.Quantity(t, ndmin=1).to(u.s).value
        E = u.Quantity(E, ndmin=1).to(u.MeV).value
        tE_grid = np.stack(np.meshgrid(t, E, indexing='ij'), axis=-1)

        initial_spectra = {}
        for flavor in ThreeFlavor:
            initial_spectra[flavor] = self.interpolation[flavor](tE_grid) / (u.MeV * u.s)
```
(Note that we also need to remove the units in between, otherwise `np.stack` will complain that we’re mixing units.)

I’ve played around with a toy example to figure out the correct order of indices (`self.interpolation[flavor](tE_grid)[i, j] == self.interpolation[flavor]((t[i], E[j]))`), but since this is so easy to mix up, I would appreciate if someone could double-check that.

JostMigenda replied:
The same applies to the `TypeIa` class.

jpkneller committed:
Update ccsn.py

Moving SNOWGLOBES model class to base.py. Will become parent to TypeIa and Wright_2017 model classes.

jpkneller committed:
Update base.py

Moved SNOwGLoBES model class here. It will become the parent for the TypeIa and Wright_2017 model classes.

jpkneller committed:
Update typeIa_loaders.py

jpkneller committed:
Update typeIa.py

jpkneller committed:
Update pisn_loaders.py

jpkneller committed:
Update model_files.yml

jpkneller committed:
Merge branch 'main' into jpkneller-pisn-models

jpkneller committed:
Update SNOwGLoBES_models.ipynb

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

Modified SNOWGLOBES class to follow suggestion by Jost

jpkneller committed:
Update typeIa.py

jpkneller committed:
Update typeIa_loaders.py

jpkneller committed:
Update TypeIa.ipynb

jpkneller committed:
Update base.py

jpkneller committed:
Update typeIa_loaders.py

jpkneller committed:
Update pisn_loaders.py

jpkneller committed:
Update base.py

jpkneller committed:
Update model_files.yml

jpkneller committed:
Update model_files.yml

jpkneller committed:
Create test_pisn_rates.py

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Update base.py

Fix units in SNOwGLoBES class

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update base.py

jpkneller committed:
Update model_files.yml

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Update test_pisn_rates.py

jpkneller committed:
Create test_typeIa_rates.py

jpkneller committed:
Merge branch 'main' into jpkneller-pisn-models

jpkneller committed:
Update ccsn_loaders.py

jpkneller committed:
Update ccsn.py
